### PR TITLE
Feature/components

### PR DESCRIPTION
**Pull request by krystianpach**

### CODE REVIEW

File: src/components/Input.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            ...errorMessageStyles
+        }}
+    >
+        {withError}
+    </Text>
+) : null
+
+export const Input: React.FunctionComponent<InputProps> = props => {
+    const { inputProps, withError, errorMessageStyles } = props
+    const withErrorStyles = withError && !Boolean(inputProps!.secureTextEntry)
```
Comment (jpudysz) [inline src/components/Input.tsx]: unsafe, otherwise cast it to proper type

File: src/components/Input.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    const { style, ...otherInputProps } = inputProps as TextInputProps
```
Comment (jpudysz) [inline src/components/Input.tsx]: as here but once and above

File: src/components/SelectPicker.tsx [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        pickerTitle: '',
+        pickerOptions: [],
+        customContainerStyles: {}
+    }
+
+    constructor(props: SelectPickerProps) {
+        super(props)
+
+        const [firstOption] = this.props.pickerOptions as Array<PickerOption>
+        const selectedOption = R.isDefined(this.props.selectedOption)
```
Comment (jpudysz) [inline src/components/SelectPicker.tsx]: would be cleaner to extract it to separate method like `getDefaultSelectedOption` + 2x `if`

File: src/components/SelectPicker.tsx [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+            selectedOption
+        }
+
+        this.confirmSelection = this.confirmSelection.bind(this)
+    }
+
+    toggleModalVisibility(isModalVisible: boolean) {
+        this.setState({
+            isModalVisible,
+            selectedOption: !isModalVisible ? this.props.selectedOption : this.state.selectedOption
```
Comment (jpudysz) [inline src/components/SelectPicker.tsx]: indentation

File: src/components/SelectPicker.tsx [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+        })
+    }
+
+    confirmSelection() {
+        if (this.props.onSuccess) {
```
Comment (jpudysz) [inline src/components/SelectPicker.tsx]: can we use `R.ifDefined`

File: src/components/SelectPicker.tsx [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+            isModalVisible: false
+        })
+    }
+
+    getLabelForSelectedOption(selectedOption?: OptionValue) {
+        if (!R.isDefined(selectedOption)) {
+            return undefined
+        }
+
+        const option = this.props.pickerOptions!.find(option => option.value === selectedOption)
```
Comment (jpudysz) [inline src/components/SelectPicker.tsx]: use casts

File: src/components/SelectPicker.tsx [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+        return option
+            ? option.label
+            : undefined
+    }
+
+    renderSelectedOptionPreview() {
+        const selectedOptionLabel = this.getLabelForSelectedOption(this.props.selectedOption)
+        const previewText = selectedOptionLabel
+            ? selectedOptionLabel
+            : this.props.placeholder || ''
```
Comment (jpudysz) [inline src/components/SelectPicker.tsx]: to default props value?

File: src/lib/common/colors.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+export const colors = {
```
Comment (jpudysz) [inline src/lib/common/colors.ts]: check if these colors are used somewhere, otherwise please remove it

File: src/types/selectPicker.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
+import { OptionValue } from './form'
+import { ViewStyle } from 'react-native'
```
Comment (jpudysz) [inline src/types/selectPicker.ts]: order

File: src/lib/types/react-native.ts [hunk 10]
```diff
@@ -0,0 +1,11 @@
+import { ImageStyle, TextStyle, ViewStyle } from 'react-native'
+
+export type Styles = {
+    [key: string]: ViewStyle & TextStyle & ImageStyle
+}
+
+export enum ModalAnimation {
```
Comment (jpudysz) [inline src/lib/types/react-native.ts]: is it used anywhere?

Comment (krystianpach) [reply src/lib/types/react-native.ts]: right now not but it will be in picker

**Pull request merged**